test/config-luatest/upgrade_test.lua

```diff
@@ -0,0 +1,108 @@
+local t = require('luatest')
+local server = require('test.luatest_helpers.server')
+local treegen = require('test.treegen')
+
+local g = t.group('config_upgrade_test')
+
+g.before_all(function()
```

Comment:
It's messy, as config tests for some reason doesn't work inside `workdir` if `wal.dir` and `snapshot.dir` is not set. So, manual tweaks are needed to copy xlogs to the same directory, where instances work, and to make luatest's netbox work.